conda_rpms/tests/unit/build_rpm_structure/test_create_rpmbuild_content.py

```diff
@@ -0,0 +1,110 @@
+        self.assertEqual(self.mock_render_env.call_args_list, expected)
+        self.mock_create_tag.reset_mock()
+        self.mock_render_env.reset_mock()
+
+    def test_basic(self):
+        ctag = 'env-{}-2017_01_01'.format(self.bname)
+        self._add_label('current.txt', ctag)
+        ntag = 'env-{}-2017_02_02'.format(self.bname)
+        self._add_label('next.txt', ntag)
+        count = self.branch.commit.count()
+
+        with self.temp_dir() as dname:
+            # Make the expected RPM build SPECS directory.
+            os.makedirs(os.path.join(dname, 'SPECS'))
+
+            # Test - no prior build, so both labels build.
+            state = {}
+            self._check_full_build(dname, state, ctag, ntag, count)
+
+            # Test - prior build of current label but mismatching tag,
+            #        so both labels build.
+            tag = 'env-{}-2018_01_01'.format(self.bname)
+            state = dict(default=dict(current=tag))
+            self._check_full_build(dname, state, ctag, ntag, count)
+
+            # Test - prior build of current label with matching tag,
+            #        so only next label builds.
+            state = dict(default=dict(current=ctag))
+            create_rpmbuild_content(self.repo, dname, self.config, state)
+            self.assertEqual(self.mock_create_tag.call_count, 1)
+            expected = [((self.repo, ntag, dname, self.config),)]
+            self.assertEqual(self.mock_create_tag.call_args_list, expected)
+            self.assertEqual(self.mock_render_env.call_count, 1)
+            expected = [((self.bname, 'next', self.config, ntag, count),)]
+            self.assertEqual(self.mock_render_env.call_args_list, expected)
+            self.mock_create_tag.reset_mock()
+            self.mock_render_env.reset_mock()
+
+            # Test - prior build of next label with matching tag,
+            #        so only current label builds.
+            state = dict(default=dict(next=ntag))
+            create_rpmbuild_content(self.repo, dname, self.config, state)
+            self.assertEqual(self.mock_create_tag.call_count, 1)
+            expected = [((self.repo, ctag, dname, self.config),)]
+            self.assertEqual(self.mock_create_tag.call_args_list, expected)
+            self.assertEqual(self.mock_render_env.call_count, 1)
+            expected = [((self.bname, 'current', self.config, ctag, count),)]
+            self.assertEqual(self.mock_render_env.call_args_list, expected)
+            self.mock_create_tag.reset_mock()
+            self.mock_render_env.reset_mock()
+
+            # Test - prior build of both labels, so no artifacts are built.
+            state = dict(default=dict(current=ctag, next=ntag))
+            create_rpmbuild_content(self.repo, dname, self.config, state)
+            self.assertEqual(self.mock_create_tag.call_count, 0)
+            self.assertEqual(self.mock_render_env.call_count, 0)
```

Comment:
@lbdreyer The problem is that if you split them out in to separate tests (as you naturally would want to do) you then have to address the problem that `setUp` is creating the `self.repo` for them to do their testing. So you'd probably want to create one test repo per test case to ensure there was no cross contamination between test cases.

Although there may be a way to setup one repo for the class that all the class test cases use ... I'll see if I can break the  tests out ...